tests/cpp/test_matmul_scheduler.cpp

```diff
@@ -2699,6 +2743,66 @@ TEST_F(MatmulSchedulerTest, DISABLED_RequireExternalPlugin) {
   MatmulParams params;
 }
 
+// Test that we can segment a Fusion containing two matmuls
+TEST_F(MatmulSchedulerTest, Llama2FFN) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(7, 5, 9, 0);
+
+  for (bool enable_fusion : {false, true}) {
+    auto fusion = std::make_unique<Fusion>();
+    FusionGuard fg(fusion.get());
+
+    auto tv0 = makeContigTensor(2, DataType::Half);
+    auto tv1 = makeContigTensor(2, DataType::Half);
+    auto tv2 = makeContigTensor(2, DataType::Half);
+    fusion->addInput(tv0);
+    fusion->addInput(tv1);
+    fusion->addInput(tv2);
+
+    auto tv3 = matmul(tv0, tv1);
+    auto tv4 = matmul(tv0, tv2);
+
+    // silu
+    auto tv5 = mul(sigmoid(tv3), tv3);
+
+    auto tv6 = mul(tv5, tv4);
+
+    fusion->addOutput(tv6);
+
+    FusionExecutorCache executor_cache(std::move(fusion));
+
+    const int M = 504, N = 136, K = 248;
+
+    at::manual_seed(0);
+    auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA);
+    auto t0 = at::randn({M, K}, options);
+    auto t1 = at::randn({K, N}, options);
+    auto t2 = at::randn({K, N}, options);
+    std::vector<c10::IValue> inputs{t0, t1, t2};
+
+    EnableOptionsGuard eog;
+    if (enable_fusion) {
+      EnableOptionsGuard::getCurOptions().set(EnableOption::FuseMatmul);
+    } else {
+      EnableOptionsGuard::getCurOptions().unset(EnableOption::FuseMatmul);
```

Comment:
We shouldn't have to unset this when using the options guard.

Reply:
This is just there in case it was set as an env var.